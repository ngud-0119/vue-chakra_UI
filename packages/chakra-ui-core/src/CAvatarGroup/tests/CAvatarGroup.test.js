import { CAvatar, CAvatarGroup } from '../..'
import { render } from '@/tests/test-utils'

const renderComponent = (props) => {
  const base = {
    components: {
      CAvatar,
      CAvatarGroup
    },
    template: `
    <CAvatarGroup max="2">
      <CAvatar
        name="Mesut Koca"
        src="https://pbs.twimg.com/profile_images/953743486842474496/cOrUdK4z_200x200.jpg"
      />
      <CAvatar
        name="Evan You"
        src="https://pbs.twimg.com/profile_images/888432310504370176/mhoGA4uj_400x400.jpg"
      />
      <CAvatar
        name="Jonathan Bakebwa"
        src="https://res.cloudinary.com/xtellar/image/upload/v1572857445/me_zqos4e.jpg"
      />
    </CAvatarGroup>`,
    ...props
  }
  return render(base)
}

it('should render correctly', () => {
  const { asFragment } = renderComponent()
  expect(asFragment()).toMatchSnapshot()
})

test('renders a number avatar showing count of truncated avatars', () => {
  const { getByText } = renderComponent()

  getByText('+1')
})
