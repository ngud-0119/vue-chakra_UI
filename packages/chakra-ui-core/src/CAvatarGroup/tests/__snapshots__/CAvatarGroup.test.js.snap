// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-80aj1h"
  >
    <div
      class="css-152ubph"
    >
      <div
        aria-label="Mesut Koca"
        class="css-1e8yzjf"
      >
        MK
      </div>
    </div>
    <div
      class="css-jq8oqj"
    >
      <div
        aria-label="Evan You"
        class="css-1e8yzjf"
      >
        EY
      </div>
    </div>
    <div
      class="css-3dare1"
    >
      +1
    </div>
  </div>
</DocumentFragment>
`;
